### PR TITLE
feat: send daily "feed" notification w/ new resources posted 📣

### LINKED ISSUE #265: Send daily Slack notification for new resources posted 📣
### Description

In order to keep members in the loop about new resources and keep the Resource Database top of mind (which can help with feature adoption), we should send a daily notification of all the latest resources that were posted.

### Acceptance Criteria

Every day at 9 AM PT, we should send a message to a Slack channel (TBD on which one it will be), which sends a list of all the resources that were posted in the last 24 hours. The message should look something like this:

```
Some new resources were posted in the Resource Database:
- [Title of Resource] by @SlackUser1
- [Title of Resource] by @SlackUser2
- [Title of Resource] by @SlackUser3

Check out these latest resources [here](/link-to-resource-database-with-filter-for-last-24-hours)!

Show some love if any of these are helpful!
```

Note: If there were no resources posted in the last 24 hours, we shouldn't send the notification.

### Additional Context

This should be implemented as a Bull job, similar to the [birthday notifications](https://github.com/colorstackorg/oyster/blob/ba2e91c5d27303f3b5414ec79ef154806c8218d1/packages/core/src/modules/member/use-cases/send-birthday-notification.ts#L8).

### PR DESCRIPTION
**Pull request by Hamza-Mos**
## Description ✏️

Closes #265

Describe what this PR does.

- Created bull run to send a daily slack notification for new resources posted within the past 24 hours
- Added a search parameter to resources page to filter by 24 hours within the timestamp passed

Manually tested search params locally as well as fetching resources.

## Type of Change 🐞

- [x] Feature - A non-breaking change which adds functionality.
- [ ] Fix - A non-breaking change which fixes an issue.
- [ ] Refactor - A change that neither fixes a bug nor adds a feature.
- [ ] Documentation - A change only to in-code or markdown documentation.
- [ ] Tests - A change that adds missing unit/integration tests.
- [ ] Chore - A change that is likely none of the above.

## Checklist ✅

- [x] I have done a self-review of my code.
- [x] I have manually tested my code (if applicable).
- [x] I have added/updated any relevant documentation (if applicable).

## Considerations

@ramiAbdou @tomas-salgado Currently the message only works for resources that contain links (essentially url-based resources). I have an approach on how to handle pdf-based resources but open to any suggestions:

- I could query for any resource-based attachments then generate a presigned url to attach to the message such as shown here: https://github.com/colorstackorg/oyster/blob/abadd7eefecc115bc024f6ddb8b667a99300b609/apps/member-profile/app/routes/_profile.resources.tsx#L109
- The problem with that is that the expiration time for the url would have to be long enough for most people to see it, which makes this approach a little fragile
- Alternatively, we could keep it this way where only url-based resources are attached as part of the message and then members can click on the link at the end and see a full list of resources that were added in the last 24 hours which should include any pdf-based resources. OR I could add pdf-based resources as well but just the title and the person that posted them (no links). 

Let me know what you think

### CODE REVIEW

File: packages/core/src/shared/env.ts [hunk 1]
```diff
@@ -21,6 +21,7 @@ export const ENV = {
   MAILCHIMP_SERVER_PREFIX: process.env.MAILCHIMP_SERVER_PREFIX as string,
   MEMBER_PROFILE_URL: process.env.MEMBER_PROFILE_URL as string,
   REDIS_URL: process.env.REDIS_URL as string,
+  RESOURCES_CHANNEL_ID: process.env.RESOURCES_CHANNEL_ID as string,
```
Comment (Hamza-Mos) [inline packages/core/src/shared/env.ts:24]: @ramiAbdou @tomas-salgado I put this as a placeholder for now - i don't think a `#resources` (or something similar) channel exists yet for this feature

**Comment by tomas-salgado** [comment]
@Hamza-Mos in response to your comment in 'Considerations', I think it's a better approach to still list all pdf-based resources with the title alongside the other resources. I'd even say it's not necessary to provide the link to URL-based resources directly in the Slack message for consistency, but if it's straightforward enough it could be good. The key is just we want to emphasize users to go to the actual Resources page.

**Comment by Hamza-Mos** [comment]
> @Hamza-Mos in response to your comment in 'Considerations', I think it's a better approach to still list all pdf-based resources with the title alongside the other resources. I'd even say it's not necessary to provide the link to URL-based resources directly in the Slack message for consistency, but if it's straightforward enough it could be good. The key is just we want to emphasize users to go to the actual Resources page.

@tomas-salgado  For some reason, it did not let me respond to your comment directly so I'm just doing it here. I ended up just adding a condition for nullable links in the message. So for pdf-based resources, a link will not be attached (since `resource.link` will be null and a pre-signed URL would have to be generated), but for URL-based resources, a link will be attached. I could remove links entirely for consistency if you think that's better but adding the links was pretty straightforward enough so I just kept it. Let me know what you think

**Review by ramiAbdou — APPROVED** [review]
@Hamza-Mos Thank you so much for implementing this -- you killed it on your first feature! 🚀 I really appreciate the way you dove in and were able to soak in the codebase style and flow so quickly!

There was some context that you didn't have at the time of creating the PR . We're actually striving to create a "feed" like experience both in the Member Profile (hopefully soon) and also in the Slack (via a channel) in general to bring more visibility to things that happen in the Member Profile. The notification you implemented was initially scoped to be just a "resource" notification, but the idea is that we can just include it in a daily "feed" notification in the Slack. That being said, the main update here is that we have a new `feed` module where this logic will live. Thinking that the next iteration of this daily "feed" notification will be including:
- Company reviews posted yesterday.
- Members that joined the directory yesterday.

That being said, I refactored your work, see [here](https://github.com/colorstackorg/oyster/pull/435/files/4c49d4eeb05627d5f8640aa15a978c89657c2902..92192fadf2422070681b5caf9e66929b8a656098). Happy to chat through any changes!

Also, I'm going to create a private channel and add you so that we can test this in production before we add everybody to the channel.

Thanks again man! 🔥

**Comment by ramiAbdou** [comment]
> > @Hamza-Mos in response to your comment in 'Considerations', I think it's a better approach to still list all pdf-based resources with the title alongside the other resources. I'd even say it's not necessary to provide the link to URL-based resources directly in the Slack message for consistency, but if it's straightforward enough it could be good. The key is just we want to emphasize users to go to the actual Resources page.
> 
> @tomas-salgado For some reason, it did not let me respond to your comment directly so I'm just doing it here. I ended up just adding a condition for nullable links in the message. So for pdf-based resources, a link will not be attached (since `resource.link` will be null and a pre-signed URL would have to be generated), but for URL-based resources, a link will be attached. I could remove links entirely for consistency if you think that's better but adding the links was pretty straightforward enough so I just kept it. Let me know what you think

Yeah I would prefer the consistency as well. Also, unfortunately we're currently only able to track views by clicking on the resource in the Member Profile -- there might be a way to do listen to requests to our Cloudflare resources, could explore in the future!

**Comment by Hamza-Mos** [comment]
> @Hamza-Mos Thank you so much for implementing this -- you killed it on your first feature! 🚀 I really appreciate the way you dove in and were able to soak in the codebase style and flow so quickly!
> 
> There was some context that you didn't have at the time of creating the PR -- we're actually striving to create a "feed" like experience both in the Member Profile (hopefully soon) and also in the Slack (via a channel) in general to bring more visibility to things that happen in the Member Profile. The notification you implemented was scoped to be just a "resource" notification, but the idea is that we can just use that as the MVP for the "daily feed" notification in the Slack. Thinking that the next iteration of this daily "feed" notification will be including:
> 
> * Company reviews posted yesterday.
> * Members that joined the directory yesterday.
> 
> That being said, I refactored your work, see [here](https://github.com/colorstackorg/oyster/pull/435/files/4c49d4eeb05627d5f8640aa15a978c89657c2902..92192fadf2422070681b5caf9e66929b8a656098). The main update here is that we have a new `feed` module where this core logic will live. Happy to chat through any changes!
> 
> Also, I'm going to create a private channel and add you so that we can test this in production before we add everybody to the channel.
> 
> Thanks again man! 🔥

Appreciate you, thanks! Changes make sense - really excited for the new feed 😤

**Pull request merged**